### PR TITLE
Signing key rotation UI

### PR DESCRIPTION
**Pull request by goodoldneon**
## Description
Create signing key rotation UI behind a feature flag

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/packages/components/src/Button/Button.tsx [hunk 1]
```diff
@@ -2,9 +2,10 @@ import type { UrlObject } from 'url';
 import React, { forwardRef } from 'react';
 import type { Route } from 'next';
 import Link from 'next/link';
+import { Tooltip, TooltipContent, TooltipTrigger } from '@inngest/components/Tooltip';
 import { IconSpinner } from '@inngest/components/icons/Spinner';
 
-import { cn } from '../utils/classNames';
+import { classNames } from '../utils/classNames';
```
Comment (anafilipadealmeida) [inline ui/packages/components/src/Button/Button.tsx:8]: The `classNames` is deprecated, we should be using `cn`. With `cn`, the classNames passed as props override the Button ones. I merged that change [yesterday](https://github.com/inngest/inngest/pull/1296/commits/ef0cac2828722a65c83640c5964c3fa2c10c9ba3), as I had to adjust some Buttons with new icons

File: ui/packages/components/src/icons/common.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export type IconProps = {
```
Comment (anafilipadealmeida) [inline ui/packages/components/src/icons/common.ts:1]: Do we still need this file?

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/manage/signing-key/page-new.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  if (activeKeys.length > 1) {
+    // Unreachable
+    throw new Error('More than one active key found');
+  }
+  const activeKey = activeKeys[0];
+  const inactiveKeys = data.filter((key) => !key.isActive);
+
+  return (
+    <div className="mb-4 mt-8 flex items-center justify-center">
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/manage/signing-key/page-new.tsx:43]:
```suggestion
    <div className="my-8 flex items-center justify-center">
```

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/manage/signing-key/page-new.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      <div className="mb-8 w-full max-w-[800px] divide-y">
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/manage/signing-key/page-new.tsx:44]:
```suggestion
      <div className="w-full max-w-[800px] divide-y">
```

File: ui/apps/dashboard/src/components/Secret/Secret.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  return (
+    <div
+      className={cn(
+        'flex rounded-md border border-slate-300 bg-slate-50 text-slate-500',
+        className
+      )}
+    >
+      <div className="flex grow items-center truncate border-r border-slate-300 p-2 text-slate-500">
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/components/Secret/Secret.tsx:30]: The `truncate` here is unnecessary

**Review by anafilipadealmeida — APPROVED** [review]

File: ui/apps/dashboard/src/components/Secret/Secret.tsx (hunk 5 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/components/Secret/Secret.tsx:30]: It doesn't work without it 😢

Comment (goodoldneon) [reply ui/apps/dashboard/src/components/Secret/Secret.tsx:30]: But `overflow-hidden` isn't needed

**Pull request merged**